### ISSUE TITLE
Error: no implicit conversion of Symbol into String

### ISSUE THREAD

**Issue by igordot** [Homebrew/brew#3721]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

# Please always follow these steps:
- [X] Confirmed this is a problem with running a `brew` command and not `brew install`ing or the post-install behaviour of one or more formulae? If it's a formulae-specific problem please file this issue at the relevant tap e.g. for Homebrew/homebrew-core https://github.com/Homebrew/homebrew-core/issues/new
- [X] Ran `brew update` and retried your prior step?
- [X] Ran `brew doctor`, fixed all issues and retried your prior step?
- [X] Ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
- What happened (include command output)
- What you expected to happen
- Step-by-step reproduction instructions (by running `brew` commands)

# Features
Please replace this section with:
- a detailed description of your proposed feature
- the motivation for the feature
- how the feature would be relevant to at least 90% of Homebrew users
- what alternatives to the feature you have considered

We will close this issue or ask you to create a pull-request if it's something we're not actively planning to work on.

I have not updated Homebrew in a few weeks. I updated today. Now, when I run `brew doctor`, I get this:
```
Warning: Calling 'depends_on :mysql' is deprecated!
Use 'depends_on "mysql"' instead.
/usr/local/Homebrew/Library/Homebrew/dependency_collector.rb:88:in `parse_spec'

Error: no implicit conversion of Symbol into String
Please report this bug:
  https://docs.brew.sh/Troubleshooting.html
/usr/local/Homebrew/Library/Homebrew/formulary.rb:363:in `extname'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:363:in `loader_for'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:280:in `factory'
/usr/local/Homebrew/Library/Homebrew/dependency.rb:33:in `to_formula'
/usr/local/Homebrew/Library/Homebrew/dependency.rb:96:in `block in expand'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/delegate.rb:341:in `each'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/delegate.rb:341:in `block in delegating_block'
/usr/local/Homebrew/Library/Homebrew/dependency.rb:83:in `expand'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1478:in `recursive_dependencies'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1500:in `missing_dependencies'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:13:in `block in missing_deps'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:12:in `each'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:12:in `missing_deps'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:869:in `check_missing_deps'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:46:in `block in doctor'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:38:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:38:in `doctor'
/usr/local/Homebrew/Library/Homebrew/brew.rb:100:in `<main>'
```
I am not sure if the warning is related to the error, but I am posting both just in case.

The `brew config` results:
```
HOMEBREW_VERSION: 1.5.1
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: ace2d569dcd3a45d0fc49a6742540b6f7a957fbc
Last commit: 14 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: cd303e56dd62467cc8f0c555dda56340be4ac4b9
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
CPU: quad-core 64-bit broadwell
Homebrew Ruby: 2.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Clang: 9.0 build 900
Git: 2.14.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Perl: /usr/bin/perl
Python: /Users/i/miniconda3/bin/python => /Users/id460/miniconda3/bin/python3.5
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Java: 1.8.0_92
macOS: 10.13.2-x86_64
Xcode: 9.2
CLT: N/A
X11: 2.7.11 => /opt/X11
```

**Comment by igordot** [comment]
I was able to solve the issue. I think the problem was with the [science tap](https://github.com/Homebrew/homebrew-science) (recently deprecated). When I removed it (`brew untap homebrew/science`), the error went away.

I am not sure if you want to consider the issue closed. There might be better ways to handle the error, such as making it more clear where it is coming from.

**Comment by alyssais** [comment]
I think we could use better messaging on this. I've seen a lot of user confusion about this error.

**Comment by MikeMcQuaid** [comment]
@alyssais Do you have a reproduction case?

**Comment by alyssais** [comment]
Ignore me, I misread the issue.

**Comment by MikeMcQuaid** [comment]
If you `brew update` do you still see this?

**Comment by igordot** [comment]
Correct, `brew update` did not fix the problem.

**Comment by ilovezfs** [comment]
@igordot he meant that you should try again, as we've released a new version of Homebrew with changes that may affect the behavior. Are you actually currently still able to reproduce the issue?

**Comment by igordot** [comment]
@ilovezfs I cannot reproduce the issue, but I was able to resolve it by untapping homebrew/science (see https://github.com/Homebrew/brew/issues/3721#issuecomment-359612223).

**Comment by ilovezfs** [comment]
@igordot right I saw that :) If you re-tap homebrew/science, does it come back?

**Comment by igordot** [comment]
I updated brew and went back to my previous taps.

Here are my updated taps:
```
caskroom/cask
homebrew/core
homebrew/science
homebrew/versions
```
This leads to `brew doctor` error:
```
Error: no implicit conversion of Symbol into String
Please report this bug:
  https://docs.brew.sh/Troubleshooting.html
/usr/local/Homebrew/Library/Homebrew/formulary.rb:363:in `extname'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:363:in `loader_for'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:280:in `factory'
/usr/local/Homebrew/Library/Homebrew/dependency.rb:33:in `to_formula'
/usr/local/Homebrew/Library/Homebrew/dependency.rb:96:in `block in expand'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/delegate.rb:341:in `each'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/delegate.rb:341:in `block in delegating_block'
/usr/local/Homebrew/Library/Homebrew/dependency.rb:83:in `expand'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1478:in `recursive_dependencies'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1500:in `missing_dependencies'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:13:in `block in missing_deps'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:12:in `each'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:12:in `missing_deps'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:869:in `check_missing_deps'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:46:in `block in doctor'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:38:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:38:in `doctor'
/usr/local/Homebrew/Library/Homebrew/brew.rb:100:in `<main>'
```

Looks like the `depends_on` warning was resolved.

I just realized my previous comment may have been incorrect. Running `brew untap homebrew/science` did not solve the error. However, tapping the replacement (`brew tap brewsci/science`) does.

Curiously, `brew tap brewsci/science` generates many warnings:
```
==> Tapping brewsci/science
Cloning into '/usr/local/Homebrew/Library/Taps/brewsci/homebrew-science'...
remote: Counting objects: 610, done.
remote: Compressing objects: 100% (607/607), done.
remote: Total 610 (delta 1), reused 271 (delta 1), pack-reused 0
Receiving objects: 100% (610/610), 546.21 KiB | 5.20 MiB/s, done.
Resolving deltas: 100% (1/1), done.
Warning: Calling 'depends_on :fortran' is deprecated!
Use 'depends_on "gcc"' instead.
/usr/local/Homebrew/Library/Taps/brewsci/homebrew-science/Formula/dotwrp.rb:16:in `<class:Dotwrp>'
Please report this to the brewsci/science tap!

Warning: Calling 'depends_on :python' is deprecated!
Use 'depends_on "python"' instead.
/usr/local/Homebrew/Library/Taps/brewsci/homebrew-science/Formula/libsbml.rb:32:in `<class:Libsbml>'
Please report this to the brewsci/science tap!

Warning: Calling 'depends_on :python' is deprecated!
Use 'depends_on "python"' instead.
/usr/local/Homebrew/Library/Taps/brewsci/homebrew-science/Formula/med-file.rb:21:in `<class:MedFile>'
Please report this to the brewsci/science tap!

Warning: Calling 'depends_on :perl' is deprecated!
Use 'depends_on "perl"' instead.
/usr/local/Homebrew/Library/Taps/brewsci/homebrew-science/Formula/masurca.rb:26:in `<class:Masurca>'
Please report this to the brewsci/science tap!

...
```

To summarize, no error with these following taps:
```
brewsci/bio
brewsci/science
caskroom/cask
homebrew/core
homebrew/versions
```

**Comment by ilovezfs** [comment]
What does `brew config` now show?

**Comment by MikeMcQuaid** [comment]
@igordot Please run `brew doctor --verbose --debug` and paste the output here, thanks!

**Comment by igordot** [comment]
To answer both questions.

With the problematic taps:
```
caskroom/cask
homebrew/core
homebrew/science
homebrew/versions
```
For `brew config`:
```
HOMEBREW_VERSION: 1.5.2
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 60a30e966b7cece5bd4823dae3fb981ab85106ea
Last commit: 7 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 58b9f4ad8dca79a35b5983c5d09e8ffa5a0cdf50
Core tap last commit: 7 minutes ago
HOMEBREW_PREFIX: /usr/local
CPU: quad-core 64-bit broadwell
Homebrew Ruby: 2.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/bin/ruby
Clang: 9.0 build 900
Git: 2.14.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Perl: /usr/bin/perl
Python: /Users/id460/miniconda3/bin/python => /Users/id460/miniconda3/bin/python3.5
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Java: 1.8.0_92
macOS: 10.13.2-x86_64
Xcode: 9.2
CLT: N/A
X11: 2.7.11 => /opt/X11
```
And `brew doctor --verbose --debug`:
```
Checking check_for_symlinked_cellar
Checking check_for_tap_ruby_files_locations
Checking check_for_unlinked_but_not_keg_only
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libxml2.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/icu4c.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/readline.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/sqlite.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gettext.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl@1.1.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/texinfo.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/openssl.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libffi.rb
Checking check_for_unsupported_macos
Checking check_git_newline_settings
Checking check_git_status
Checking check_git_version
Checking check_homebrew_prefix
Checking check_if_xcode_needs_clt_installed
Checking check_multiple_cellars
Checking check_path_for_trailing_slashes
Checking check_ruby_version
Checking check_ssl_cert_file
Checking check_tmpdir
Checking check_tmpdir_sticky_bit
Checking check_user_curlrc
Checking check_user_path_1
Checking check_user_path_2
Checking check_user_path_3
Checking check_which_pkg_config
Checking check_xcode_8_without_clt_on_el_capitan
Checking check_xcode_license_approved
Checking check_xcode_minimum_version
Checking check_xcode_prefix
Checking check_xcode_prefix_exists
Checking check_xcode_select_path
Checking check_xcode_up_to_date
Checking check_xquartz_up_to_date
Checking check_for_broken_symlinks
Checking check_missing_deps
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pkg-config.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/proj.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libtiff.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gmp.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/lzlib.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/wget.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/json-c.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libidn2.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pandoc.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libpng.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/hugo.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pixman.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libgeotiff.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/freetype.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpfr.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libunistring.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/glib.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gcc.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/cairo.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/xz.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/fontconfig.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/htop.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/giflib.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/gsl.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/bedtools.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-science/kent-tools.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::FormulaLoader): loading /usr/local/opt/kent-tools/.brew/kent-tools.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mysql.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/shellcheck.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/isl.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/freexl.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/pcre.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/dos2unix.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/node.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/jpeg.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/protobuf.rb
/usr/local/Homebrew/Library/Homebrew/brew.rb (Formulary::TapLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libmpc.rb
Error: no implicit conversion of Symbol into String
Please report this bug:
  https://docs.brew.sh/Troubleshooting.html
/usr/local/Homebrew/Library/Homebrew/formulary.rb:363:in `extname'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:363:in `loader_for'
/usr/local/Homebrew/Library/Homebrew/formulary.rb:280:in `factory'
/usr/local/Homebrew/Library/Homebrew/dependency.rb:33:in `to_formula'
/usr/local/Homebrew/Library/Homebrew/dependency.rb:96:in `block in expand'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/delegate.rb:341:in `each'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/delegate.rb:341:in `block in delegating_block'
/usr/local/Homebrew/Library/Homebrew/dependency.rb:83:in `expand'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1478:in `recursive_dependencies'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1500:in `missing_dependencies'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:13:in `block in missing_deps'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:12:in `each'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:12:in `missing_deps'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:869:in `check_missing_deps'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:46:in `block in doctor'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:38:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:38:in `doctor'
/usr/local/Homebrew/Library/Homebrew/brew.rb:100:in `<main>'
```

**Comment by MikeMcQuaid** [comment]
Can you run `export HOMEBREW_NO_AUTO_UPDATE=1` temporarily and can you add `p ref` on line 363 of `/usr/local/Homebrew/Library/Homebrew/formulary.rb` and then rerun `brew doctor --verbose --debug`? Thanks!

**Comment by byronformwalt** [comment]
@MikeMcQuaid I followed your procedure and determined that on my machine, `:rbenv` is the offending value of `ref` on line 363.  

I was able to resolve the issue by inserting the following line before line 363:
`ref = ref.to_s if ref.is_a? Symbol`

**Comment by apjanke** [comment]
@igordot Triggering this might be dependent on the specific formulae you have installed. Could you post the output of `brew ls` for us?

**Comment by igordot** [comment]
Here is my `brew ls`:
```
bedtools	glib		kent-tools	lzlib		pkg-config
cairo		gmp		libffi		mpfr		proj
dos2unix	gsl		libgeotiff	mysql		protobuf
fontconfig	htop		libidn2		node		readline
freetype	hugo		libmpc		openssl		shellcheck
freexl		icu4c		libpng		openssl@1.1	sqlite
gcc		isl		libtiff		pandoc		texinfo
gettext		jpeg		libunistring	pcre		wget
giflib		json-c		libxml2		pixman		xz
```

**Comment by MikeMcQuaid** [comment]
> When I removed it (brew untap homebrew/science), the error went away.

Given that seems to fix it and the rest of this is a mystery: closing this out. If @byronformwalt or others still have an unresolved issue here: please shout and I'll reopen. Thanks all!

**Comment by byronformwalt** [comment]
@Mike McQuaid

Will do.  I don’t recall which formula caused the problem for me, but it was a simple fix.  If it happens again, I’ll post here.

**Comment by reidpr** [comment]
I ran into this issue as well. The patch above does hide the error.

The offending formula seems to be `emacs`. `brew info emacs`:

    emacs: stable 25.3 (bottled), devel 26.0.91, HEAD
    GNU Emacs text editor
    https://www.gnu.org/software/emacs/
    /usr/local/Cellar/emacs/25.3 (4,051 files, 119.6MB) *
      Built from source on 2017-09-18 at 08:54:52 with: --with-cocoa

`brew cleanup emacs`, which removed versions 24.5, 25.1, and 25.2, did not help.

Uninstalling `emacs` and reinstalling with `brew cask install emacs` as recommended did eliminate the error message. I don't yet know if this version meets my needs, but I'll try it out.

**Comment by MikeMcQuaid** [comment]
> The patch above does hide the error for me.

@reidpr Which patch?

**Comment by reidpr** [comment]
This patch:

> I was able to resolve the issue by inserting the following line before line 363:
> ref = ref.to_s if ref.is_a? Symbol
